### PR TITLE
Teach RedisBlobAccess to update TTL during Get and FindMissing calls

### PR DESCRIPTION
**Pull request by asuffield**
This makes the key_ttl feature be LRU-like instead of FIFO.

### CODE REVIEW

File: pkg/blobstore/redis_blob_access.go [hunk 1]
```diff
@@ -26,18 +26,20 @@ type redisBlobAccess struct {
 	readBufferFactory  ReadBufferFactory
 	digestKeyFormat    digest.KeyFormat
 	keyTTL             time.Duration
+	refreshOnUse       bool
```
Comment (EdSchouten) [inline pkg/blobstore/redis_blob_access.go:29]: refreshTTLOnUse?

**Review by EdSchouten — COMMENTED** [review]
To me it isn't really clear what the use of TTLs in Redis is in our case. Isn't it just sufficient to set the eviction policy to LRU on the server globally?

I read the Redis source code some time ago. As far as I know, EXISTS and GET operations also affect the eviction policy state as expected.

**Comment by asuffield** [comment]
Interesting question, I will have to run performance tests to find out. If correct, we could maybe remove keyTTL entirely.

**Comment by asuffield** [comment]
Some experimentation later, I have concluded that redis's own LRU is at least as good as this, so keyTTL is an antifeature. I'll follow up by removing it.

**Pull request closed without merging**